src/langchain_google_alloydb_pg/embeddings.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO: Remove below import when minimum supported Python version is 3.10
+from __future__ import annotations
+
+import json
+from typing import List
+
+from langchain_core.embeddings import Embeddings
+from sqlalchemy import text
+
+from .engine import AlloyDBEngine
+
+
+class AlloyDBEmbeddings(Embeddings):
+    """Google AlloyDB Embeddings available via Model Endpoint Management."""
+
+    def __init__(self, engine: AlloyDBEngine, model_id: str):
```

Comment:
The engine is required for model validation and generation of the embedding being returned to the user (vs in DB only generation). However, I'm not sure I like the default experience of requiring the engine. This could be the default:
```python
embedding_service = AlloyDBEmbeddings() # Default model_id = textembedding_gecko
embedding_service = AlloyDBEmbeddings(model_id='textembedding-gecko@001')
```
vs having to specify the engine:
```python
embedding_service = AlloyDBEmbeddings(engine)
embedding_service = AlloyDBEmbeddings(model_id='textembedding-gecko@001', engine=engine)
```

Reply:
Talked offline and the engine experience is not bad.